### PR TITLE
Implement GRPC interceptor for Zipkin

### PR DESCRIPTION
**Pull request by lnshi**
- Implement GRPC interceptor for Zipkin.

- Addressed [Issue #73](https://github.com/openzipkin/zipkin-js/issues/73).

### CODE REVIEW

**Comment by adriancole** [comment]
Thanks for raising this. Looks like most of the pieces are here. I'm wondering if you could package this into a shimmer module, perhaps with patchClient and patchServer functions like https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/master/src/plugins/plugin-grpc.js

This would be easier for folks to use as it would have less instructions

File: packages/zipkin-instrumentation-grpc/src/ZipkinGrpcInterceptor.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    let ctxSampled = grpcMetadataFromIncomingCtx.get('x-b3-sampled')[0];
+    if (!(ctxSampled in ['0', '1'])) {
+      ctxSampled = '0';
+    }
+
+    const ctxTraceInfo = new TraceId({
+      traceId: new Some(ctxTraceId),
+      parentId: new Some(ctxParentId),
+      spanId: ctxSpanId,
+      sampled: new Some(ctxSampled)
```
Comment (adriancole) [inline packages/zipkin-instrumentation-grpc/src/ZipkinGrpcInterceptor.js]: per httpServer.js we shouldn't assume Some (ex if the header is not there). you can check for that one's impl as it is more correct

Comment (wdittmer) [reply packages/zipkin-instrumentation-grpc/src/ZipkinGrpcInterceptor.js]: @lnshi 
Created a pull request that 'should' fix that: https://github.com/lnshi/zipkin-js/pull/1/commits/0931720f0aa1a170ad66ce35b52c949ca9c4544c

**Comment by tomasAlabes** [comment]
Hi! @adriancole @lnshi is there a plan to merge this PR?

**Comment by adriancole** [comment]
there's no plan to merge this pull request as-is as it has a bug and has gone dormant. If someone wants to raise a derived pull request, please do.

If someone wants to give it a retry, please simplify the README as currently it is hard to understand and includes instructions for golang which make things blurry. If you derive code from here, please credit @lnshi and try to simplify the setup steps as possible

**Pull request closed without merging**